lib/ansible/modules/network/aci/msc_label.py

```diff
@@ -0,0 +1,180 @@
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.aci.msc import MSCModule, msc_argument_spec, issubset
+
+
+def main():
+    argument_spec = msc_argument_spec()
+    argument_spec.update(
+        label=dict(type='str', required=False, aliases=['name', 'label_name']),
+        label_id=dict(type='str', required=False),
+        display_name=dict(type='str'),
+        type=dict(type='str', default='site', choices=['site']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['label']],
+            ['state', 'present', ['label']],
+        ],
+    )
+
+    label = module.params['label']
+    label_id = module.params['label_id']
+    label_type = module.params['type']
+    state = module.params['state']
+
+    msc = MSCModule(module)
+
+    path = 'labels'
+
+    # Query for existing object(s)
+    if label_id is None and label is None:
+        msc.existing = msc.query_objs(path)
+    elif label_id is None:
+        msc.existing = msc.get_obj(path, displayName=label)
+        if msc.existing:
+            label_id = msc.existing['id']
+    elif label is None:
+        msc.existing = msc.get_obj(path, id=label_id)
+    else:
+        msc.existing = msc.get_obj(path, id=label_id)
+        existing_by_name = msc.get_obj(path, displayName=label)
+        if existing_by_name and label_id != existing_by_name['id']:
+            msc.fail_json(msg="Provided label '{1}' with id '{2}' does not match existing id '{3}'.".format(label, label_id, existing_by_name['id']))
+
+    # If we found an existing object, continue with it
+    if label_id:
+        path = 'labels/{id}'.format(id=label_id)
+
+    if state == 'query':
+        pass
+
+    elif state == 'absent':
+        msc.previous = msc.existing
+        if msc.existing:
+            if module.check_mode:
+                msc.existing = {}
+            else:
+                msc.existing = msc.request(path, method='DELETE')
+
+    elif state == 'present':
+        msc.previous = msc.existing
+
+        msc.sanitize(dict(
+            id=label_id,
+            displayName=display_name,
+            type=label_type,
+        ), collate=True)
+
+        if msc.existing:
+            if not issubset(msc.sent, msc.existing):
+                if module.check_mode:
+                    msc.existing = msc.proposed
+                else:
+                    msc.existing = msc.request(path, method='PUT', data=msc.sent)
+        else:
+            if module.check_mode:
+                msc.existing = msc.proposed
+            else:
+                msc.existing = msc.request(path, method='POST', data=msc.sent)
```

Comment:
Should proposed be updated?

Reply:
...changed...should changed be updated?

Reply:
Yes, that's what `msc.request()` is doing.